src/Client.js

```diff
@@ -0,0 +1,14 @@
+import SocketService from './core/SocketService';
+import ChainManager from './core/ChainManager';
+
+export default class Client {
+  constructor() {
+    this._socketService = new SocketService();
```

Comment:
My initial reasoning for this is was to use two manager classes to do the handling of chains and snapshots. At the minute, I decided to add all snapshot handling to chainManager as there is not much in the way of using snapshots at the minute but this may change.
I can easily add the socketService to be an internal implementation of chainManager.